asllib/error.ml

```diff
@@ -227,36 +228,41 @@ module PPrint = struct
     | BadReturnStmt None ->
         pp_print_text f
           "ASL Typing error: cannot return something from a procedure."
-    | UnexpectedSideEffect s -> fprintf f "Unexpected side-effect: %s" s
-    | UncaughtException s -> fprintf f "Uncaught exception: %s" s
+    | UnexpectedSideEffect s -> fprintf f "Unexpected side-effect: %s." s
+    | UncaughtException s -> fprintf f "Uncaught exception: %s." s
     | OverlappingSlices slices ->
         fprintf f "ASL Typing error:@ overlapping slices@ @[%a@]." pp_slice_list
           slices
     | BadLDI ldi ->
         fprintf f "Unsupported declaration:@ @[%a@]." pp_local_decl_item ldi
     | BadRecursiveDecls decls ->
-        fprintf f "ASL Typing error:@ multiple recursive declarations:@ @[%a@]"
+        fprintf f "ASL Typing error:@ multiple recursive declarations:@ @[%a@]."
           (pp_comma_list (fun f -> fprintf f "%S"))
           decls
     | UnrespectedParserInvariant -> fprintf f "Parser invariant broke."
     | ConstrainedIntegerExpected t ->
         fprintf f
-          "ASL Typing error:@ constrained@ integer@ expected,@ provided@ %a"
+          "ASL Typing error:@ constrained@ integer@ expected,@ provided@ %a."
           pp_ty t
     | ParameterWithoutDecl s ->
         fprintf f
           "ASL Typing error:@ explicit@ parameter@ %S@ does@ not@ have@ a@ \
-           corresponding@ defining@ argument"
+           corresponding@ defining@ argument."
           s
     | BaseValueEmptyType t ->
-        fprintf f "ASL Execution error: base value of empty type %a" pp_ty t
+        fprintf f "ASL Typing error: base value of empty type %a." pp_ty t
+    | BaseValueNonStatic (t, e) ->
+        fprintf f
+          "ASL Typing error:@ base@ value@ of@ type@ %a@ containing@ \
```

Comment:
I'm not sure the term "non-static value" makes sense.
How about something like "base value of type %a cannot be statically determined since it consists of %a"?